### PR TITLE
Deprecate form-data "size" in POST /artifacts requests

### PR DESCRIPTION
**Pull request by alfrunes**
This PR removes the size parameter from the POST request as it's considered redundant (we can't necessarily verify size ahead of time). The parameter is still allowed, but ignored.

### CODE REVIEW

File: s3/filestorage.go [hunk 1]
```diff
@@ -219,63 +217,80 @@ func (s *SimpleStorageService) Exists(ctx context.Context, objectID string) (boo
...
+	//			},
+	//		},
+	//	}
+	//	if _, err := s.client.PutObjectTagging(input); err != nil {
+	//		l := log.FromContext(r.Context())
+	//		l.Warnf("failed to tag artifact : %s\n", objectID)
+	//	}
+	//}
+
+	//return nil
```
Comment (merlin-northern) [inline s3/filestorage.go:293]: this we do not need I presume?

File: api/http/api_deployments.go [hunk 2]
```diff
@@ -402,42 +409,67 @@ func (d *DeploymentsApiHandlers) GenerateImage(w rest.ResponseWriter, r *rest.Re
...
-	multipartUploadMsg.MetaConstructor.Description = r.FormValue("description")
-
-	sizeValue := r.FormValue("size")
-	var size int64
-	var err error
-	if sizeValue != "" {
-		size, err = strconv.ParseInt(sizeValue, 10, 64)
+	var sizeParsed bool
+	for {
+		part, err := r.NextPart()
```
Comment (merlin-northern) [inline api/http/api_deployments.go:418]: I have never done that in go, so please excuse me asking: so we do nothing with the `part` thing?

File: s3/filestorage.go (hunk 1 above)
Comment (alfrunes) [reply s3/filestorage.go:293]: Oops...

File: app/app.go [hunk 3]
```diff
@@ -251,6 +263,25 @@ func (d *Deployments) handleArtifact(ctx context.Context,
...
+				l.Errorf(
+					"Failed to delete broken image %s: %s",
+					artifactID, err)
+			}
+		}()
+		return "", errors.Wrap(
+			ErrInvalidArtifactSize,
+			"size smaller than expected",
+		)
+	}
```
Comment (merlin-northern) [inline app/app.go]: discussed on slack

File: app/app.go [hunk 4]
```diff
@@ -276,10 +248,17 @@ func (d *Deployments) handleArtifact(ctx context.Context,
...
-			ErrInvalidArtifactSize,
-			"size smaller than expected",
-		)
+		return "", ErrModelArtifactFileTooLarge
+	}
+	// close the pipe
+	pW.Close()
+
+	// Assign artifact size to the actual uploaded size
+	multipartUploadMsg.ArtifactSize = MaxImageSize - lr.N
```
Comment (merlin-northern) [inline app/app.go:257]: are we *sure* here that `MaxImageSize>=lr.N` ?

Comment (alfrunes) [reply app/app.go:257]: Yes, the `else if` above catches that case.

**Comment by alfrunes** [comment]
Superseded by #406

**Pull request closed without merging**